torchdrivesim/traffic_controls.py

```diff
@@ -22,13 +27,31 @@ class BaseTrafficControl:
         mask: BxN boolean tensor indicating whether a given traffic control element is present and not a padding.
     """
     def __init__(self, pos: Tensor, allowed_states: Optional[List[str]] = None,
-                 replay_states: Optional[Tensor] = None, mask: Optional[Tensor] = None):
+                 replay_states: Optional[Tensor] = None, mask: Optional[Tensor] = None, ids: Optional[List] = None,
+                 use_mock_lights: bool = False, preset_states: Optional[Dict[int, List[str]]] = None):
         self.pos = pos
+        self.ids = ids
+        self.use_mock_lights = use_mock_lights
         self.allowed_states = allowed_states if allowed_states is not None else self._default_allowed_states()
         self.replay_states = replay_states if replay_states is not None else self._default_replay_states()
         self.mask = mask if mask is not None else self._default_mask()
+        self.preset_states = preset_states
+        self.virtual_overlap = None
 
+        self.pos[...,2] = 1
         self.corners = box2corners_th(self.pos)
+
+        moved_dist = 5
+        new_pos = pos.clone()
+        for i in range(pos.shape[0]):
+          for j in range(pos.shape[1]):
+            dx = moved_dist * math.cos(pos[i][j][-1])
+            dy = moved_dist * math.sin(pos[i][j][-1])
+            new_pos[i][j][0] = pos[i][j][0] - dx
+            new_pos[i][j][1] = pos[i][j][1] + dy
+
+        self.moved_corners = box2corners_th(new_pos)
```

Comment:
You should extend the `__init__` function of the stop sign class to define them since these only are relevant there.